### DISCUSSION TITLE
Seq underutilizing VM resources.

### DISCUSSION THREAD

**Discussion by jmolla31** [datalust/seq-tickets#2038] [Help]
Hi! We've recently upgraded our seq deployment to 2023.4 and also upscaled the VM (4 Core 16GB Windows Server on Azure) it is being hosted on.

Seq is currently the only service running on that machine and we never see memory usage go above 30-40%. I would expect the service to be able to use as much RAM as possible to speed up queries.

Is this maybe a misconfiguration on our side or expected behaviour?

The only memory-related setting we have set is: `process.workingSetLimitBytes=16000000000`

Thanks!

**Comment by KodrAus** [comment]
Hi @jmolla31 👋 

This is a change we made in 2023.4; instead of holding on to file-backed memory internally Seq will eagerly release it to the OS when it's finished reading a file. This makes Seq's working set appear much smaller, but doesn't mean all that memory isn't in a useful state. The OS will typically keep those file backed pages resident for the next time we read them, but will also freely drop them if the memory is requested by another process.

We ended up doing this because Windows doesn't like operating close to 100% RAM usage, and in Docker that can risk being OOM-killed by the host.

**Comment by nblumhardt** [comment]
I'd also recommend running without the working set limit if possible"

```
seq config clear -k process.workingSetLimitBytes
seq service restart
```

Working set limits aren't usually necessary and can have unpredictable side-effects.

HTH!
